# PREFACE



# Introduction

=======================================

Kisi-Kisi LC 3

Nama    : Rafina Dhiya Pradani

Batch   : HCK-21

Program ini dibuat untuk melakukan analisis statistik deskriptif dan API pada Perusahaan peminjaman sepeda di San Fransisco

=======================================

# ANSWER

In [15]:
import pandas as pd
from scipy import stats
import matplotlib.pyplot as plt
import seaborn as sns
from IPython.core.display import display, HTML

In [16]:
# Melakukan SetUp koneksi BigQuery ke Google Colab
from google.colab import auth
from google.cloud import bigquery
auth.authenticate_user()
print('Authenticated')

project_id = "afi-ftds-hacktiv8" #Menggunakan project_id masing-masing
client = bigquery.Client(project=project_id)

Authenticated


In [36]:
# Menampilkan data dari BigQuery dengan memanggil variabel df
df = client.query('''
SELECT duration_sec
FROM `bigquery-public-data.san_francisco_bikeshare.bikeshare_trips`
LIMIT 3000;
''').to_dataframe()

df

duration_sec
0             1113
1              788
2              917
3              473
4              580
...            ...
2995           964
2996           444
2997           671
2998          1441
2999           624

[3000 rows x 1 columns]

# STATISTIK DESKRIPTIF

## 1. Perhitungan central tendency (mean, median, dan modus)

In [37]:
# Menghitung mean
print('Duration sec in Average:',df['duration_sec'].mean(),'sec')

# Menghitung median
print('Median of Duration Sec:',df['duration_sec'].median(),'sec')

# Menghitung modus
print('Mode of Duration sec:',df['duration_sec'].mode().values[0],'sec')

Duration sec in Average: 976.386 sec
Median of Duration Sec: 498.0 sec
Mode of Duration sec: 234 sec


Mean / rata-rata durasi perjalanan customer adalah 976.386 sec
Median durasi perjalanan customer adalah 498.0 sec
Modus durasi perjalanan customer adalah 234 sec

## 2. Pengecekan dispersi (standar deviasi, variance, dan range)

In [38]:
# Pengecekan standar deviasi
std = df['duration_sec'].std()
print('Standard Deviation of duration sec:',std)

Standard Deviation of duration sec: 4057.199757011408


Standar deviasi pada duration sec adalah 4057.199757011408

In [39]:
# Pengecekan variance
variance = df['duration_sec'].var()
print('Variance of duration sec:',variance)

Variance of duration sec: 16460869.868293429


Variance dari data duration sec adalah 16460869.868293429

In [40]:
# Pengecekan range
range = df['duration_sec'].max() - df['duration_sec'].min()
print('Range of duration sec:',range)

Range of duration sec: 85910


Range pada duration sec adalah 85910

## 3. Pengecekan skewness dan kurtosis

In [42]:
# Pengecekan skewness
print(f"Skewness: {df['duration_sec'].skew()}")

Skewness: 14.139091375363616


nilai skewness adalah 14.139091375363616

In [43]:
# Pengecekan kurtosis
print(f"Kurtosis: {df['duration_sec'].kurtosis()}")

Kurtosis: 224.14896554962405


Nilai kurtosis adalah 224.14896554962405

## 4. Menganalisa apakah ada data outlier berdasarkan sebaran data

- Berdasarkan normality test skewness yang dilakukan, menunjukkan angka 14.139091375363616 (high skew) sehingga terdapat outlier.
- Berdasarkan normality test kurtosis yang dilakukan, menunjukkan angka 224.14896554962405 (leptokurtic type) sehingga terdapat outlier.

## 5. Cari tahu berapa persentase outlier terhadap data keseluruhan.

In [51]:
# Mengecek nilai Q1, Q3, dan IQR
q1 = df['duration_sec'].quantile(0.25)
q3 = df['duration_sec'].quantile(0.75)
iqr = q3-q1

# Mengecek nilai batas atas dan bawah
up_bound = q3 + 1.5*iqr
low_bound = q1 - 1.5*iqr

print(f'Upper Boundary: {up_bound:.2f}' )
print(f'Lower Boundary: {low_bound:.2f}')

Upper Boundary: 1472.88
Lower Boundary: -388.12


Upper boundary pada duration sec adalah 1472.88
lower bounday pada duration sec adalah -388.12

In [55]:
# Outlier
outlier = df['duration_sec'][(df['duration_sec']<low_bound) | (df['duration_sec']>up_bound)]

# Normal
no_outlier = df['duration_sec'][(df['duration_sec']>=low_bound) & (df['duration_sec']<=up_bound)]

print(f'Percentage of outliers: {len(outlier)*100/len(df["duration_sec"]):.2f}%')

Percentage of outliers: 6.00%


Persentase outlier terhadap data secara keseluruhan adaah 6.00%

## 6. Bagaimana perubahan pemusatan data sebelum dan sesudah penghapusan outlier?

In [57]:
# Mengecek perubahan pemusatan data sebelum penghapusan outlier
print("Sebelum Extreme Value Analysis")
print('Average:',df['duration_sec'].mean())
print('Median :',df['duration_sec'].median())
print('Mode   :',df['duration_sec'].mode().values[0])
print("-"*30)

# Mengecek perubahan pemusatan data setelah penghapusan outlier
print("Sesudah Extreme Value Analysis")
print('Average:',no_outlier.mean())
print('Median :',no_outlier.median())
print('Mode   :',no_outlier.mode().values[0])

Sebelum Extreme Value Analysis
Average: 976.386
Median : 498.0
Mode   : 234
------------------------------
Sesudah Extreme Value Analysis
Average: 536.4195035460992
Median : 471.0
Mode   : 234
